File: vacation_backend/Application/DTOs/Vacation/VacationRequestHistoryDto.cs
using vacation_backend.Domain.Enums;

namespace vacation_backend.Application.DTOs.Vacation
{
    public class VacationRequestHistoryDto
    {
        public int Id { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int TotalDays { get; set; }
        public VacationType VacationType { get; set; }
        public VacationStatusEnum Status { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? LastModifiedBy { get; set; }
        public DateTime? LastModifiedAt { get; set; }
        public string? ApprovedBy { get; set; }
        public DateTime? ApprovedAt { get; set; }


    }
}

File: vacation-frontend/src/models/vacation/VacationRequestHistoryDto.ts
export interface VacationRequestHistoryDto {
  id: number
  startDate: string
  endDate: string
  totalDays: number
  vacationType: number
  status: number
  createdBy: string
  createdAt: string
  lastModifiedBy?: string | null
  lastModifiedAt?: string | null
  approvedBy?: string | null
  approvedAt?: string | null
}